uk.co.compucorp.civicrm.civihrsampledata/tests/phpunit/CRM/CiviHRSampleData/Importer/AbsencePeriodTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+require_once EXTENSION_ROOT_DIR . 'CRM/CiviHRSampleData/Importer/AbsencePeriod.php';
+
+/**
+ * Class CRM_CiviHRSampleData_Importer_AbsencePeriodTest
+ *
+ * @group headless
+ */
+class CRM_CiviHRSampleData_Importer_AbsencePeriodTest extends CRM_CiviHRSampleData_BaseImporterTest {
+
+  private $rows;
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->install('org.civicrm.hrabsence')
+      ->apply();
+  }
+
+  public function setUp() {
+    $this->rows = [];
+    $this->rows[] = $this->importHeadersFixture();
+  }
+
+  public function testImport() {
+    $this->rows[] = [
+      '2016',
+      '2016 (Jan 1 to Dec 31)',
+      '2016-01-01 02:00:00',
+      '2017-01-01 01:59:59',
+    ];
+
+    $this->runImporter('CRM_CiviHRSampleData_Importer_AbsencePeriod', $this->rows);
+
+    $this->assertEquals('2016', $this->apiQuickGet('HRAbsencePeriod','name', '2016'));
```

Comment:
Sorry, there is a typo on my first comment. It should say "Also, you should at least assert the other fields".

For example, in that test you're asserting only that you got the expected name. But you should, test that you have the expected values for all the other fields. This will give you some assurance that, at least for this minimal set of data, the parsing/validation is right.

Here is an example of a breaking change that this test will not catch: Currently, you're using the API to import the data. Your test will pass because the API automatically handles the date format conversion and will, in the end, pass it to the BAO in YmdHis format. Now, imagine that, for some reason, you have to stop using the API and call the BAO directly. It will not receive the date in the correct format and maybe it might truncate it and save the wrong information on the database base, but your test will still pass because it's only checking the name.

Yeah, I know it's quite a lot of tests but it's part of the job. We already found a few bugs just by reviewing the code. It wouldn't be a surprise to find a few more when we actually start using this. Tests can help us find these faster and also fix them faster when they appear (yeah, we'll have bugs even if we do a lot of tests). So time should no be an excuse to not write tests. Also, when something it's hard to test it might be a sign that the code itself can be improved to be more flexible and easier to test (can't say if this is the case here though)

Anyway, I agreed we should meet at the middle for this specific case. So I think it will be good enough if you add assertion to all the properties instead of checking only for one of them.